_tutorials/fr/2024-03-19-chromatic/steps/other-tests.md

```diff
@@ -0,0 +1,255 @@
+
+export const Default: Story = {
+  play: async ({ canvasElement }) => {
+
+    ...
+
+    const closeButton = getByRole('button', { name: "Fermer la modale" });
+    expect(closeButton).toHaveFocus();
+  }
+}
+```
+
+**Les interactions sont vertes, le test passe bien !** On aura même une double vérification avec le snapshot de Chromatic qui va montrer l'état de focus par défaut du navigateur sur le bouton de fermeture.
+
+[Voici le lien de documentation de `byRole`, avec toutes les options possibles](https://testing-library.com/docs/queries/byrole) et [la documentation de `jest-dom`](https://github.com/testing-library/jest-dom)
+
+### Tester les événements
+
+On peut tester les events émis par les composants **avec les actions de Storybook**. Les événements émis **s’affichent dans l’onglet Actions** mais sont bien utilisables dans la fonction `play`. On peut vérifier si **un événement a bien été émis lors d’une interaction, et avec quelle valeur**, s'il en a.
+
+On voudrait vérifier que notre composant Modale émet bien un évènement **à l'ouverture de la modale**. On a déjà un `onOpenModal` dans notre composant, c'est lui que nous allons écouter. Dans les `argTypes` de `meta`, dans la story, nous allons ajouter cette configuration&nbsp;:
+
+```ts
+const meta = {
+  title: 'Example/Modal',
+  component: Modal,
+  argTypes: { onOpenModal: { action: 'onOpenModal' } }, // <= nouvelle ligne ici
+  parameters: {
+    layout: 'centered',
+  },
+  tags: ['autodocs'],
+} satisfies Meta<typeof Modal>;
+```
+
+Maintenant lorsque l'interaction se lance on a dans l'onglet `Actions` une entrée `onOpenModal` ! Il ne nous reste plus qu'à tester que **l'action est bien effectuée** dans notre fonction `play`&nbsp;:
+
+```ts
+export const Default: Story = {
+  play: async ({ canvasElement, args }) => { // On ajoute `args` qui contient onOpenModal
+    const { getByRole } = within(canvasElement);
+
+    expect(args.onOpenModal).not.toHaveBeenCalled(); // On vérifie que onOpenModal n'a pas encore été appelée
+
+    const openButton = getByRole('button', { name: "Ouvrir la modale" });
+    await userEvent.click(openButton);
+
+    const closeButton = getByRole('button', { name: "Fermer la modale" });
+    expect(closeButton).toHaveFocus();
+
+    expect(args.onOpenModal).toHaveBeenCalled(); // On vérifie que onOpenModal a été appelée
+  },
+};
+```
+
+Si tout s'est bien passé **les tests sont toujours verts !**
+
+### Conditionner le lancement des interactions
+
+Lorsqu’on ajoute une interaction, **elle est systématiquement jouée à l’ouverture** de la story. Ce n’est pas toujours le comportement souhaité. Il est possible de ne **lancer les interactions que pour Chromatic**, et également de masquer certains éléments à Chromatic pour le snapshot avec la fonction `isChromatic`. La fonction renverra `true` si on est sur Chromatic, sinon `false`. On peut l’utiliser directement dans les stories ou dans la fonction `Play`.
```

Comment:
par soucis d'homogénéité avec ce qui a été mis plus haut.

```suggestion
Lorsqu’on ajoute une interaction, **elle est systématiquement jouée à l’ouverture** de la story. Ce n’est pas toujours le comportement souhaité. Il est possible de ne **lancer les interactions que pour Chromatic**, et également de masquer certains éléments à Chromatic pour le snapshot avec la fonction `isChromatic`. La fonction renverra `true` si on est sur Chromatic, sinon `false`. On peut l’utiliser directement dans les stories ou dans la fonction `play`.
```

Reply:
c'est modifié